cmd/thanos/rule.go

```diff
@@ -564,7 +555,10 @@ func runRule(
 		}
 
 		router.WithPrefix(webRoutePrefix).Post("/-/reload", func(w http.ResponseWriter, r *http.Request) {
-			reload <- struct{}{}
+			if err := reloadRules(logger, ruleFiles, ruleMgr, evalInterval, metrics); err != nil {
```

Comment:
Hm... so why we wait on reloadSignal above if we just reload stuff here? :thinking:

Reply:
@bwplotka 
> reload <- struct{}{}

This line is removed.

Reply:
See my comment above.

Reply:
I don't think we need to have separate err message for HTTP and signal (:

Reply:
concrete error message helps troubleshooting.

Reply:
Ok, then let's maybe use another channel?